### DISCUSSION TITLE
USB ports are blocked after using Ntrip

### DISCUSSION THREAD

**Discussion by jehlijos** [tomojitakasu/RTKLIB#732] [Q&A]
Hi, I am trying to use Ntrip correction data with my reciever. I collect NMEA messages from u-blox via USB port with python script (PySerial). But if I start str2str or rtkrcv fucntion It stops the skripts communication with USB port.

Is there any way to set rtkrcv.conf to make this work?
If you are interested how my script looks like, check it here :) https://github.com/jehlijos/GNSS-EGNOS-EDAS-public_copy

Thanks for any suggestions.

**Comment by DavidKelleySCSC** [comment]
Just a few thoughts after a quick code walk.

- You never close the USB port, perhaps this is why you can not restart it.  
- You have no provision for a bad NTRIP reply (failed connection) and that can cause the NTRIP Caster to get upset after repeated bad connections, at times blocking you.
- You seem to be decoding the lat/long in the format of DD.ddd (degrees and fractions) when in fact it is encoded as DDMM.SSsss (degrees, minutes, seconds), so you need to account for the minutes field. 

Again, this was just a quick look.

**Comment by jehlijos** [comment]
Thanks a lot!
So after adding condition for bad NTRIP reply and if NTRIP workes properly without any errors, it should work fine ?
